### ISSUE TITLE
copy pasted SVG or PNG cannot be undone

### ISSUE THREAD

**Issue by Mrazator** [excalidraw/excalidraw#8321]
**Reproduction**

1. create a rectangle and trigger it's context menu (with right-click)
2. select "Copy to clipboard as PNG" or "Copy to clipboard as SVG"
3. paste (i.e. with `cmd + v`)
4. undo (i.e. with `cmd + z`)

**Expected behaviour**

PNG or SVG is removed from the scene

**Actual behaviour**

Nothing is undone (undo stack is likely not populated on paste)

**Comment by Lokendra-sinh** [comment]
@Mrazator I believe it's working fine for me. 

create a rectangle
copy the rectangle as svg
paste the rectangle ctrl+v cmd+v
hit undo and yeah it does work smoothly.

Let me know if I'm missing something.

**Comment by Mrazator** [comment]
Can't confirm, still not working as expected.

https://github.com/user-attachments/assets/366d1c89-ad93-4a5f-b37c-70999bc6dd39

**Comment by mtolmacs** [comment]
Adding some notes as I think this is a good first issue. This is likely an issue with recording the paste from clipboard in the history store. You can investigate these places to get started:
* `packages/excalidraw/components/App.tsx` - `App::pasteFromClipboard()`
* `packages/excalidraw/store.ts` for the history debugging: `Store::commit()`

**Comment by vedang685** [comment]
@mtolmacs Is this Issue up for grab?

**Comment by mtolmacs** [comment]
@vedang685 yes, but you can see if an issue is taken from the comments (or a PR is linked).

Please send a PR I can link to the ticket (even if it's draft).

**Comment by vedang685** [comment]
@mtolmacs PR #8582 Created the pr successfully that has resolved this bug
